Simplify exponents fails with controlled operations

There is an [assert statement](https://github.com/unitaryfund/mitiq/blob/master/mitiq/utils.py#L67) to check that the operation is a `cirq.GateOperation`, which a `cirq.ControlledOperation` is not. I don't think the function should fail because there may be operations which are not gate operations. The better behavior is to only try to simplify `GateOperation`s, i.e., skip `ControlledOperation`s and other not gate operations. 

Example:

```python
import cirq
from mitiq.mitiq_qiskit.conversions import to_qiskit

circuit = cirq.Circuit(
    cirq.ControlledGate(cirq.SWAP, num_controls=1).on(*cirq.LineQubit.range(3))
)
print(circuit)
# 0: ───@───
#       │
# 1: ───×───
#       │
# 2: ───×───

to_qiskit(circuit)
```

raises

```python
mitiq/mitiq/mitiq/utils.py in _simplify_circuit_exponents(circuit)
     65         # Iterate over operations in moment
     66         for op in moment:
---> 67             assert isinstance(op, GateOperation)
     68 

AssertionError: 
```

--- a/mitiq/tests/test_utils.py
+++ b/mitiq/tests/test_utils.py
@@ -23,6 +23,7 @@ import cirq
 from cirq import (
     LineQubit,
     Circuit,
+    ControlledGate,
     X,
     Y,
     Z,
@@ -221,6 +222,16 @@ def test_simplify_gate_exponent(gate):
 def test_simplify_gate_exponent_with_gates_that_cannot_be_simplified(gate):
     # Check the gate is not simplified (same representation)
     assert _simplify_gate_exponent(gate).__repr__() == gate.__repr__()
+
+
+def test_simplify_circuit_exponents_controlled_gate():
+    circuit = Circuit(
+        ControlledGate(CNOT, num_controls=1).on(*LineQubit.range(3))
+    )
+    copy = circuit.copy()
+
+    _simplify_circuit_exponents(circuit)
+    assert _equal(circuit, copy)
 
 
 def test_simplify_circuit_exponents():

--- a/mitiq/utils.py
+++ b/mitiq/utils.py
@@ -64,11 +64,16 @@ def _simplify_circuit_exponents(circuit: Circuit) -> None:
         simplified_operations = []
         # Iterate over operations in moment
         for op in moment:
-            assert isinstance(op, GateOperation)
+
+            if not isinstance(op, GateOperation):
+                simplified_operations.append(op)
+                continue
+
             if isinstance(op.gate, EigenGate):
                 simplified_gate: Gate = _simplify_gate_exponent(op.gate)
             else:
                 simplified_gate = op.gate
+
             simplified_operation = op.with_gate(simplified_gate)
             simplified_operations.append(simplified_operation)
         # Mutate the input circuit

